pkg/crc/cluster/cluster.go

```diff
@@ -32,7 +32,7 @@ const (
 	KubeletServerCert = "/var/lib/kubelet/pki/kubelet-server-current.pem"
 	KubeletClientCert = "/var/lib/kubelet/pki/kubelet-client-current.pem"
 
-	AggregatorClientCert = "/etc/kubernetes/static-pod-resources/kube-apiserver-certs/configmaps/aggregator-client-ca/ca-bundle.crt"
+	AggregatorClientCert = "/etc/kubernetes/static-pod-resources/kube-apiserver-certs/secrets/aggregator-client/tls.crt"
```

Comment:
The commit log says
> During testing of 4.17.0.ec.1 found out that `secrets/aggregator-client/tls.crt` is regenerated as part of auto cert rotation before `configmaps/aggregator-client-ca/ca-bundle.crt`.

Is `configmaps/aggregator-client-ca` really regenerated after `secrets/aggregator-client`? Or does this happen in the opposite order?

Reply:
As a preparatory commit, I'd remove https://github.com/crc-org/crc/blob/2fa7958844be56353ebfd6c104281a77a5a24b5c/pkg/crc/machine/start.go#L605-L626 as this change has an impact on this code.

Reply:
Yes it is really regenerated after `secrets/aggregator-client`, I ssh'ed to the node and checked this.

Reply:
It's quite odd, as it would make more sense if `secrets/aggregator-client` used the new CA, not an older one.
If `configmaps/aggregator-client-ca` is the last to be regenerated, why do we change to check for `secrets/aggregator-client`? When `aggregator-client-ca` is fully renewed, `aggregator-client` should also be fine?

Reply:
> As a preparatory commit, I'd remove 

Removed as first commit

>It's quite odd, as it would make more sense if secrets/aggregator-client used the new CA, not an older one.
If configmaps/aggregator-client-ca is the last to be regenerated, why do we change to check for secrets/aggregator-client? When aggregator-client-ca is fully renewed, aggregator-client should also be fine?

During that internal conversation figure out the due to certsyncer controller it might happen that cert files sync on the disk is out of order, added more context in the commit log.

Reply:
>  I ssh'ed to the node and checked this.

With `ls -al`, or with something like `$ cat /etc/kubernetes/static-pod-resources/kube-apiserver-certs/secrets/aggregator-client/tls.crt | openssl x509 -noout -enddate` ? I assumed you used `openssl` to check this, but seeing the updated commit log and one of your slack comments, I'm thinking you might have checked with `ls` instead?

Reply:
I checked with openssl also and with openssl, timestamp is as expected but with ls it is not because certsyncer is handled by different controller which I mentioned in the commit log.
```
During testing of 4.17.0.ec.1 found out that
`secrets/aggregator-client/tls.crt` is synced before `configmaps/aggregator-client-ca/ca-bundle.crt` which is due to the certsyncer controller.
```

Reply:
I'd be worried if the openssl timestamps were out of order. The timestamps of the files on disk are less relevant, especially with the explanation you have now.

Reply:
>  in our code we check the file 

We check it's modified/accessed/created time with `ls` / `stat` / ... ? I could only find use of `openssl x509 ...`

Reply:
> We check it's modified/accessed/created time with `ls` / `stat` / ... ? I could only find use of `openssl x509 ...`

we check the cert expiry of the file using `openssl` command only and  this command run against  `/etc/kubernetes/static-pod-resources/kube-apiserver-certs/configmaps/aggregator-client-ca/ca-bundle.crt` file,  during testing this file is not updated with rotated cert before `/etc/kubernetes/static-pod-resources/kube-apiserver-certs/secrets/aggregator-client/tls.crt` one hence this change.

Reply:
>Can we tell somehow to the certsyncer controller it should flush the certs to disk? 

@cfergeau it is the controller which is not under our control and during the internal conversation around it looks like we can't do much around it.

>  Adjusting our code to whatever it flushes to disk first feels like something which could change/break anytime, and need more changes, so it would be better if we could tell it we want to work from a well-known state on disk?

Same feeling but looks like chicken and egg situation because it is updated to etcd first (which is source of truth) as part of configmap but since the api is not working until cert is rotated so we can't put code to pull it :(